process-engine/process_engine_core
# 🐛 Fix unhandled Promise-rejection for execution paths created at ErrorBoundaryEvents

## Changes

1. Fix unhandled Promise-Rejection for Promise-chains created by `ErrorBoundaryEvents`.
2. Fix ReceiveTaskHandler interruption
3. Add missing `onProcessTerminated` notifications
4. Add test for the `onProcessTerminated` notification

## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/346
Part of https://github.com/process-engine/process_engine_runtime/issues/387

PR: #305

## How to test the changes

- Run [this ProcessModel](https://github.com/process-engine/process_engine_core/files/3733781/ConfiguredFaultyProcess.bpmn.zip), which enforces an execution path through an ErrorBoundaryEvent, towards an ErrorEndEvent
- See that the ProcessInstance finishes as expected 
- See that the ProcessInstance will no longer be listet as active
